web/tls_config.go

```diff
@@ -66,6 +68,75 @@ func (t *TLSConfig) SetDirectory(dir string) {
 	t.ClientCAs = config_util.JoinDir(dir, t.ClientCAs)
 }
 
+// VerifyPeerCertificate will check the SAN entries of the client cert if there is configuration for it
+func (t *TLSConfig) VerifyPeerCertificate(rawCerts [][]byte, _ [][]*x509.Certificate) error {
+	// sender cert comes first, see https://www.rfc-editor.org/rfc/rfc5246#section-7.4.2
+	cert, err := x509.ParseCertificate(rawCerts[0])
+	if err != nil {
+		return fmt.Errorf("error parsing client certificate: %s", err)
+	}
+
+	// Build up a slice of strings with all Subject Alternate Name values
+	sanValues := append(cert.DNSNames, cert.EmailAddresses...)
+
+	for _, ip := range cert.IPAddresses {
+		sanValues = append(sanValues, ip.String())
+	}
+
+	for _, uri := range cert.URIs {
```

Comment:
This is not a SAN value

Reply:
My reference here was that both [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280) and [Go x509 source](https://github.com/golang/go/blob/a8ca653892a470daf065b7669061d6ce1b4ff2e9/src/crypto/x509/x509.go#L753-L759) indicate that DNS and URIs are valid Subject Alternate Names.